Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -410,19 +720,18 @@ template <typename TInputImage>
 void
 ContourExtractor2DImageFilter<TInputImage>::FillOutputs()
 {
-  this->SetNumberOfIndexedOutputs(m_Contours.size());
-  int i = 0;
-  for (auto it = m_Contours.begin(); it != m_Contours.end(); it++, i++)
+  this->SetNumberOfIndexedOutputs(m_OutputNumber + m_Contours.size());
```

Comment:
Instead of increasing capacity with each label processed, the new approach allocates capacity under the assumption that each unexplored label will produce one contour.  For example, with 1000 labels, we will allocate 1000 outputs up front.  If the 970th label would take us to 1001 outputs then we will increase capacity to 1001 + 29 at that point, where 29 is the number of labels to be processed after the 970th label.  And so on each time we need more than the existing capacity.